feature_engine/timeseries/forecasting/expanding_window_features.py

```diff
@@ -224,3 +253,30 @@ def _get_new_features_name(self) -> List:
         ]
 
         return feature_names
+
+    def _agg_expanding_window_features(
+        self,
+        grouped_df: pd.core.groupby.generic.DataFrameGroupBy,
+    ) -> Union[pd.Series, pd.DataFrame]:
+        """generate expanding window features based on groups
+        Parameters
+        ----------
+        grouped_df : pd.core.groupby.generic.DataFrameGroupBy
+            dataframe of groups
```

Comment:
what does dataframe of groups mean? which groups?

Reply:
the `group_df` is the original dataframe but grouped by one or more categorical variables like (product id, store, governorate, city).

so, here we use this dataframe to loop over each single group and create the features.